src/uhs/atomizer/shard/controller.cpp

```diff
@@ -63,6 +63,19 @@ namespace cbdc::shard {
             return atomizer_handler(std::forward<decltype(pkt)>(pkt));
         });
 
+        constexpr auto max_wait = 3;
+        for(size_t i = 0; i < max_wait && m_shard.best_block_height() < 1;
+            i++) {
+            m_logger->info("Waiting to sync with atomizer");
```

Comment:
@metalicjames I have one last question on this one (and then, I think I'm ready for it to be merged): would it be simpler / more stable / more intuitive to use `condition_variable`s and `wait_for`? The result should be about the same, but is pretty easy to generalize.

Reply:
Maybe, but we don't have access to the internal state of the shard here, so I'm not sure how that could be implemented without changing the shard interface?